### PR TITLE
Convert Naming Styles to a grid-based UI and enable rules by default

### PR DESCRIPTION
**Pull request by dpoeschl**
This is the same as the previously approved #14514 with a [one-line change to not run on generated code](https://github.com/dotnet/roslyn/pull/14634/commits/21c20ac9944fc7100fac7d65f347abc2553733a5).

### CODE REVIEW

**Comment by dpoeschl** [comment]
test VSI please

**Comment by dpoeschl** [comment]
retest windows_release_unit64_prtest please
Failure: https://ci.dot.net/job/dotnet_roslyn/job/dev15-rc/job/windows_release_unit64_prtest/168/

**Comment by dpoeschl** [comment]
Retest windows_vsi_p0_open_prtest please
Failure: https://ci.dot.net/job/Private/job/dotnet_roslyn-internal/job/dev15-rc/job/windows_vsi_p0_open_prtest/6/

**Comment by dpoeschl** [comment]
retest windows_release_unit64_prtest please

**Comment by dpoeschl** [comment]
Retest windows_vsi_p0_open_prtest please

**Comment by dpoeschl** [comment]
retest windows_debug_unit64_prtest please

**Comment by dpoeschl** [comment]
VSI Tests are passing, just waiting for windows_debug_unit64_prtest to be happy (failures look unrelated to my change)

**Comment by dpoeschl** [comment]
Tagging @Pilchie and @CyrusNajmabadi who approved of the change the first time, and @ManishJayaswal who approved it for ask mode last time, though I know @mattgertz is doing that now.

**Comment by dpoeschl** [comment]
@CyrusNajmabadi approved in-person.

File: src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyles/NamingStyleDiagnosticAnalyzerBase.cs [hunk 1]
```diff
@@ -40,6 +40,7 @@ internal abstract class NamingStyleDiagnosticAnalyzerBase : DiagnosticAnalyzer,
 
         public override void Initialize(AnalysisContext context)
         {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
```
Comment (Pilchie) [inline src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyles/NamingStyleDiagnosticAnalyzerBase.cs]: As we discussed before - can we audit the rest of our style analyzers?

**Comment by dpoeschl** [comment]
retest windows_vsi_p0_open_prtest

**Comment by dpoeschl** [comment]
retest windows_vsi_p0_open_prtest please

File: src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyles/NamingStyleDiagnosticAnalyzerBase.cs (hunk 1 above)
Comment (dpoeschl) [inline src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyles/NamingStyleDiagnosticAnalyzerBase.cs]: Yep. Tracked by https://github.com/dotnet/roslyn/issues/14638 but should be easy to do.

**Comment by dpoeschl** [comment]
retest windows_debug_unit64_prtest please

**Pull request merged**